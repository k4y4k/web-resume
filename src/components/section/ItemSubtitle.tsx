import 'twin.macro'
import * as React from 'react'

interface ItemSubtitleTypes {
  subtitle: string | undefined
}

const ItemSubtitle = ({ subtitle }: ItemSubtitleTypes): JSX.Element | null => {
  if (subtitle === '') return null

  return (
    <span data-testid='itemSubtitle' tw='inline-block'>
      {subtitle}
    </span>
  )
}

export default ItemSubtitle
