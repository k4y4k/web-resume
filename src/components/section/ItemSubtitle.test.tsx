import * as React from 'react'
import { render, screen } from '@testing-library/react'
import ItemSubtitle from './ItemSubtitle'

describe('<ItemSubtitle />', () => {
  test('handles no data', () => {
    render(<ItemSubtitle subtitle='' />)

    const subtitle = screen.queryByTestId('itemSubtitle')
    expect(subtitle).not.toBeInTheDocument()
    expect(subtitle).toMatchSnapshot()
  })

  test('displays subtitle', () => {
    render(<ItemSubtitle subtitle='Yes, really! Really really!' />)

    const subtitle = screen.getByTestId('itemSubtitle')
    expect(subtitle).toHaveTextContent('Yes, really! Really really!')
    expect(subtitle).toMatchSnapshot()
  })
})
