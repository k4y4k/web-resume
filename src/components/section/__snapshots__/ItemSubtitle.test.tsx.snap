// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemSubtitle /> displays subtitle 1`] = `
.c0 {
  display: inline-block;
}

<span
  class="c0"
  data-testid="itemSubtitle"
>
  Yes, really! Really really!
</span>
`;

exports[`<ItemSubtitle /> handles no data 1`] = `null`;
